hail/python/hailtop/utils/utils.py

```diff
@@ -112,6 +112,8 @@ def grouped(n, ls):
         group = ls[:n]
         ls = ls[n:]
         yield group
+        if n < 1:
+           raise ValueError ('invalid value for n')
```

Comment:
I like the spirit of this, but it won't work as written. Try copying this definition into `ipython` and trying it out. You'll find it doesn't error when you pass the argument `0` for the parameter `n`.

There's a really powerful technique for reasoning about programs that I call "stepping through the program". To "step" a program, you pretend to be the computer. Below, I give an example of stepping through the Python expression `grouped(0, [1,2,3,4,5,6])`.

Note that instead of raising an error when I call `grouped(0, [1,2,3,4,5,6])`, I just return from the function without any error!

---

### Step 1
#### Variables
```
```
(none)
#### current line of code
```
grouped(0, [1,2,3,4,5,6])
```

---

### Step 2
#### Variables
```
n = 0
ls = [1,2,3,4,5,6]
```
#### current line of code
```
while len(ls) != 0
```

---

### Step 3
#### Variables
```
n = 0
ls = [1,2,3,4,5,6]
```
#### current line of code
```
while 6 != 0
```

---

### Step 4
#### Variables
```
n = 0
ls = [1,2,3,4,5,6]
```
#### current line of code
We reached the end of the function, so there's not really a particular "line". We'll return from grouped without erroring!

Reply:
Fixed the function